src/helper/file_helper.rs

```diff
@@ -54,30 +54,37 @@ impl FileHelper {
          directory_helper.update(&parent_directory)
     }
 
-    /// Helper function to Update content of a file in a directory listing
-    /// A writer object is returned, through which the data for the file can be written to the network
-    /// The file is actually saved in the directory listing only after `writer.close()` is invoked
-    pub fn update(&self,
-                  file            : ::file::File,
-                  mode            : ::helper::writer::Mode,
-                  parent_directory: ::directory_listing::DirectoryListing) -> Result<::helper::writer::Writer, ::errors::NfsError> {
-        let _ = try!(parent_directory.find_file(file.get_name()).ok_or(::errors::NfsError::FileNotFound));
-        Ok(::helper::writer::Writer::new(self.client.clone(), mode, parent_directory, file))
-    }
-
     /// Updates the file metadata.
     /// Returns Option<parent_directory's parent>
     pub fn update_metadata(&self,
-                           mut file        : ::file::File,
-                           user_metadata   : Vec<u8>,
-                           parent_directory: &mut ::directory_listing::DirectoryListing) -> Result<Option<::directory_listing::DirectoryListing>, ::errors::NfsError> {
-        let _ = try!(parent_directory.find_file(file.get_name()).ok_or(::errors::NfsError::FileNotFound));
-        file.get_mut_metadata().set_user_metadata(user_metadata);
+                  file            : ::file::File,
+                  parent_directory: &mut ::directory_listing::DirectoryListing) -> Result<Option<::directory_listing::DirectoryListing>, ::errors::NfsError> {
+        let _ = try!(parent_directory.find_file_by_id(file.get_id()).ok_or(::errors::NfsError::FileNotFound));
+        match parent_directory.find_file(file.get_name()) {
```

Comment:
Doing a double finds for all the cases is inefficient. It can be replaced with more efficient algorithm something like this:
```
{
    let existing_file = try!(parent_directory.find_file_by_id(file.get_id()).ok_or(::errors::NfsError::FileNotFound));
    if existing_file.get_name() != file.get_name() &&          // ie., it is a rename operation BUT
       parent_directory.find_file(file.get_name()).is_some { // a file with similar name already exists
        return Err(::errors::NfsError::NameAlreadyTaken)
    }
}
```